python/pyarrow/compat.py

```diff
@@ -165,4 +165,49 @@ def get_socket_from_fd(fileno, family, type):
         return socket.socket(fileno=fileno, family=family, type=type)
 
 
+try:
+    # This function is available after numpy-0.16.0.
+    # See also: https://github.com/numpy/numpy/blob/master/numpy/lib/format.py
+    from numpy.lib.format import descr_to_dtype
```

Comment:
Why not simply use `pickle`?
Also `descr_to_dtype` looks like a private API (it's not in the docs).

Reply:
Using `pickle` would cause compatibility issues (incompatible with former `.str` implementation). The serialization/deserialization should work between different version of pyarrow, IMO.

The `dtype_to_descr`, the counterpart of  `descr_to_dtype `, has been there for more than 10 years. The `descr_to_dtype` was added about 9 months ago. It's ok to implement our own version of this utility (the functionality of this utility is rather trivial), but I think it should be ok to try to import it from numpy, since there has already been one, though it is an undocumented API.

Reply:
Fair enough then :-)